File: app/dynamicProgramming/python/binominalCoefficient.py
import argparse

parser = argparse.ArgumentParser()
parser.add_argument('--n')
parser.add_argument('--k')
args = parser.parse_args()


def binomialCoeff(n, k):

    if k == 0 or k == n:
        return 1

    # Recursive Call
    return binomialCoeff(n-1, k-1) + binomialCoeff(n-1, k)


# Driver Program to test ht above function
n = int(args.n)
k = int(args.k)
print "Value of C(%d,%d) is (%d)" % (n, k, binomialCoeff(n, k))

# This code is contributed by Nikhil Kumar Singh (nickzuck_007)

File: app/dynamicProgramming/c++/binominalCoefficient.cpp
// A Naive Recursive Implementation
#include <stdio.h>
#include <iostream>

// Returns value of Binomial Coefficient C(n, k)
int binomialCoeff(int n, int k)
{
    // Base Cases
    if (k == 0 || k == n)
        return 1;

    // Recur
    return binomialCoeff(n - 1, k - 1) + binomialCoeff(n - 1, k);
}

/* Driver program to test above function*/
int main(int argc, char **argv)
{
    int n = atoi(argv[1]);
    int k = atoi(argv[2]);
    printf("Value of C(%d, %d) is %d ", n, k, binomialCoeff(n, k));
    return 0;
}
